contrib/seekable_format/zstd_seekable.h

```diff
@@ -0,0 +1,140 @@
+#ifndef SEEKABLE_H
+#define SEEKABLE_H
+
+#if defined (__cplusplus)
+extern "C" {
+#endif
+
+static const unsigned ZSTD_seekTableFooterSize = 9;
+
+#define ZSTD_SEEKABLE_MAGICNUMBER 0x8F92EAB1
+
+#define ZSTD_SEEKABLE_MAXCHUNKS 0x8000000U
+
+/* 0xFE03F607 is the largest number x such that ZSTD_compressBound(x) fits in a 32-bit integer */
+#define ZSTD_SEEKABLE_MAX_CHUNK_DECOMPRESSED_SIZE 0xFE03F607
```

Comment:
Could we get a static assert on this value ?

`ZSTD_compressBound()` was recently changed, it may be changed again in the future.
In case of a future change, it's necessary to ensure that `ZSTD_compressBound(ZSTD_SEEKABLE_MAX_CHUNK_DECOMPRESSED_SIZE)` doesn't overflow.

Note that it may require to expose a macro version of `ZSTD_compressBound()`